src/cloud_provider/service.rs

```diff
@@ -250,32 +251,32 @@ pub struct DatabaseOptions {
 }
 
 #[derive(Eq, PartialEq)]
-pub enum DatabaseType<'a> {
-    PostgreSQL(&'a DatabaseOptions),
-    MongoDB(&'a DatabaseOptions),
-    MySQL(&'a DatabaseOptions),
-    Redis(&'a DatabaseOptions),
+pub enum DatabaseType {
+    PostgreSQL,
+    MongoDB,
+    MySQL,
+    Redis,
 }
 
-impl<'a> ToString for DatabaseType<'a> {
+impl ToString for DatabaseType {
```

Comment:
OPT: I know it's not part of this change, might be a FAR

Usually better to implement Display for enum / struct.
We should not implement [ToString](http://doc.rust-lang.org/std/string/trait.ToString.html) manually. `ToString` trait is already implemented for all types which implement `fmt::Display`.

Implementing `Display`, `to_string()` will be available on struct / enum automatically.

Reply:
yeah, I saw it too, but I let that for another time :;(